### ISSUE TITLE
DynamoDB GSI timeouts

### ISSUE THREAD

**Issue by melbit-anthonyw** [terraform-providers/terraform-provider-aws#5471]
### Terraform Version

0.11.3

### Affected Resource(s)

aws_dynamodb_table

### Terraform Configuration Files

```hcl
resource "aws_dynamodb_table" "order" {
  name           = "${var.name}-${var.env}-orders"
  read_capacity  = "${var.read_capacity}"
  write_capacity = "${var.write_capacity}"
  hash_key       = "key"

  attribute {
    name = "key"
    type = "S"
  }

  attribute {
    name = "order_hash"
    type = "S"
  }

  attribute {
    name = "user_id"
    type = "S"
  }

  attribute {
    name = "internal_order_status"
    type = "N"
  }

  global_secondary_index {
    name               = "order_hash-index"
    hash_key           = "order_hash"
    write_capacity     = "${var.order_hash-index_write_capacity}"
    read_capacity      = "${var.order_hash-index_read_capacity}"
    projection_type    = "INCLUDE"
    non_key_attributes = ["aoo_site_id", "last_updated", "user_id"]
  }

  global_secondary_index {
    name               = "user_id-internal_order_status-index"
    hash_key           = "user_id"
    range_key          = "internal_order_status"
    write_capacity     = "${var.user_id-internal_order_status-index_write_capacity}"
    read_capacity      = "${var.user_id-internal_order_status-index_read_capacity}"
    projection_type    = "INCLUDE"
    non_key_attributes = ["last_updated", "order_status", "refund_processed", "order_response", "feedback_received", "post_time", "is_paid"]

    timeouts {
      create = "60m"
      update = "60m"
    }
  }

  timeouts {
    create = "60m"
    update = "60m"
  }

  point_in_time_recovery {
    enabled = "${var.point_in_time_recovery}"
  }
}
```

### Expected Behavior

Terraform updates the global secondary index with a new non_key_attribute and adheres to timeout settings

### Actual Behavior

aws_dynamodb_table.order: Error waiting for DynamoDB GSI "user_id-internal_order_status-index" to be created: timeout while waiting for state to become 'ACTIVE' (last state: 'CREATING', timeout: 10m0s)

Terraform updates the global secondary index with a new non_key_attribute but it takes longer than 10 minutes to create the index so Terraform times out. A subsequent run of the apply and it wants to do it all over again

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Initially created this table and the indexes but needed to add another key to the index. Terraform appears to delete and create the index with the new key however it times out at the 10 minute mark. I read the documentation and the resource should support specifying a timeout block to overwrite the timeouts for create and update but putting those blocks in either the aws_dynamodb_table or global_secondary_index blocks doesn't change the behavior.

**Comment by bflad** [comment]
Pull request submitted to use the `update` customizable timeout for each GSI update: #7453

**Comment by bflad** [comment]
The `aws_dynamodb_table` resource has been updated with the following:
* Use the `update` customizable timeout for each individual GSI update
* Increase the default `update` customizable timeout from 10 minutes to 60 minutes

The above will be released in version 1.59.0 of the Terraform AWS Provider, likely middle of this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.59.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1590-february-14-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
